### PR TITLE
Add: merge multistep and apply_every logic

### PR DESCRIPTION
**Pull request by celiolarcher**
This PR inserts the transform.apply_every logic into MultiStep wrapper, aiming to reduce memory requirements when performing gradient accumulation.

Related to issue #472

### CODE REVIEW

File: optax/_src/wrappers.py [hunk 1]
```diff
@@ -382,39 +382,28 @@ def update(self,
...
-      mid_updates = jax.tree_util.tree_map(
-          lambda sd: jnp.zeros(sd.shape, sd.dtype), updates_shape_dtype)
+      emit = state.mini_step == (k_steps - 1)
       new_state = MultiStepsState(
-          mini_step=numerics.safe_int32_increment(state.mini_step),
-          gradient_step=state.gradient_step,
-          inner_opt_state=state.inner_opt_state,
-          acc_grads=acc_grads,
+          mini_step=numerics.safe_int32_increment(state.mini_step) % k_steps,
+          gradient_step=state.gradient_step + emit,
```
Comment (hbq1) [inline optax/_src/wrappers.py:396]: gradient_step = emit * numerics.safe_int32_increment(state.gradient_step) + (1 - emit)  * state.gradient_step
